compiler/optimizer/LiveVariableInformation.cpp

```diff
@@ -477,11 +499,55 @@ void TR_LiveVariableInformation::visitTreeForLocals(TR::Node *node, TR_BitVector
          else
             {
             if (traceLiveVarInfo())
-               traceMsg(comp(), "            first reference\n");
+               traceMsg(comp(), "            first reference, ");
+
+            bool otherReferencesStillLive = false;
 
-            // if this is a load, clear it in liveCommonedLoads if we've been asked to track them
+            // if this is a load, and there are no other live commoned loads of the local,
+            // clear it in liveCommonedLoads if we've been asked to track them
             if (_liveCommonedLoads != NULL && local != NULL)
-               _liveCommonedLoads->reset(localIndex);
+               {
+               if (_seenCommonedNodeForLoadOfLocal != NULL)
+                  {
+                  // If we've encountered a commoned reference to this node, decrease
+                  // the number of distinct commoned references of the local that are
+                  // are still live.  If there are no more live commoned references to
+                  // the local, remove it from _liveCommonedLoads
+                  //
+                  if (_seenCommonedNodeForLoadOfLocal->contains(node))
+                     {
+                     int32_t numDistinctCommonedLoads = _distinctCommonedLoads[localIndex];
+                     _distinctCommonedLoads[localIndex] = numDistinctCommonedLoads - 1;
+
+                     _seenCommonedNodeForLoadOfLocal->remove(node);
+
+                     if (numDistinctCommonedLoads == 1)
+                        {
+                        _liveCommonedLoads->reset(localIndex);
+                        }
+                     else
+                        {
+                        otherReferencesStillLive = true;
+                        }
+                     }
+                  }
+               else
+                  {
+                  _liveCommonedLoads->reset(localIndex);
+                  }
+               }
+
+            if (traceLiveVarInfo())
+               {
+               if (otherReferencesStillLive)
+                  {
+                  traceMsg(comp(), "but other commoned references to this sym are still live\n");
+                  }
+               else
+                  {
+                  traceMsg(comp(), "and this was the first of all commoned references to this sym\n");
```

Comment:
This message might be reported on a node that does not have a symbol reference.  I will correct that to remove the mention of a "sym" here.